app/controllers/application_controller.rb

```diff
@@ -5,7 +5,11 @@ class ApplicationController < ActionController::Base
   before_action :authenticate_user!
 
   def after_sign_in_path_for(_resource)
```

Comment:
Let's remove this customization and add different roots for logged in and logged out user and add `before_action :check_profile_completion`
in `config/routes.rb`:
```
  authenticated :user do
    root 'dashboard#index', as: :authenticated_root
  end

  root 'pages#index'
```
in `app/controllers/application_controller.rb`:

```
before_action :check_profile_completion
...
def check_account_completion
  redirect_to new_profile_path unless current_profile.present?
end
```